### ISSUE TITLE
[major] [bug] Gradle projects add test compile path in the launcher configuration

### ISSUE THREAD

**Issue by gayanper** [microsoft/java-debug#507]
Run the attached project, and when checking the arg file, we can see that the `bin/test` is added into class path which cause issues when there are different resource files for testing purpose in applications.

[micrometer-demo.zip](https://github.com/eclipse-jdtls/eclipse.jdt.ls/files/12454816/micrometer-demo.zip)

```
-cp "/Users/home/micrometer-demo/bin/main:/Users/home/micrometer-demo/bin/test:...
```

**Comment by gayanper** [comment]
Related bug originally reported in https://github.com/eclipse-jdtls/eclipse.jdt.ls/issues/2819

**Comment by gayanper** [comment]
I have checked how the Eclipse UI generate the run command. It seems when Exclude test code is used, it uses the output path, bin/default and when Exclude test code is not used, both bin/main and bin/test are added. 

And in the extension scenario we end using all 3 paths always. When debugging it was found that bin/test comes as part of bin/test/generated-sources/annotations 's output location.

Either we can report this to buildship to add support in this scenario to provide correct paths as in Eclise or we can filter the output paths looking and the project raw classpath's output locations.

**Comment by jdneo** [comment]
Hey @gayanper

We recently just released a new tool to support Gradle project - Build Server for Gradle, maybe that can help solve your problem.

If you are interested, please follow the steps below:

- install the latest pre-release of Gradle for Java extension
  ![Picture1](https://github.com/microsoft/java-debug/assets/6193897/3fcb6b41-5fed-4cfc-8b6a-3c82f057a889)
- (Optional) If you are using VS Code stable, set the setting `"java.gradle.buildServer.enabled"` to `"on"`. Skip this if you are using Insiders
- Run `Java: Clean Java Language Server Workspace` > `Reload and Delete`
- After reload, if you still see 
![image](https://github.com/microsoft/java-debug/assets/6193897/b5055501-845a-4191-9c5c-0b53a9d8a10c)
in the Java Projects explorer, it indicates that the project is still imported by Buildship, you need to clean and reload again. (This is caused by some issues that the bundle is not registered for the first time in a new workspace)
- You should see logs
![image](https://github.com/microsoft/java-debug/assets/6193897/447c5fb5-3117-4130-a7e5-a8af743b7a61)
 in those two output channel if the new importer is working.

**Comment by gayanper** [comment]
Yes that seems to fix this issue, So I think we can close this if this build-server is the future for gradle projects, But if we are keeping the possibility to use buildship I think we might need to find away to resolve this.

**Comment by jdneo** [comment]
The long-term plan is to leverage bs approach to support the Gradle project. But since currently we are still in the exploring phase for the bs approach, maybe keep this issue open until we make the decision to use BS to replace Buildship in VS Code.

**Comment by hasanlovef** [comment]
...
